File: update.py
import json

data_path = 'c:\\Users\\menti\\Desktop\\novo-shard\\WeaponCalc\\Weapon_Analysis_Report.json'
with open(data_path, 'r', encoding='utf-8-sig') as f:
    weapons = json.load(f)

weapons.sort(key=lambda w: w['Name'])
md_content = '# Weapon Analysis Report\n\n'
categories = ['Swords', 'Maces', 'Fencing', 'Bows', 'Throwing']

for cat in categories:
    cat_weapons = [w for w in weapons if w['Skill'] == cat]
    if not cat_weapons: continue
    
    md_content += f'## {cat}\n\n'
    
    for layer in ['OneHanded', 'TwoHanded']:
        layer_weapons = [w for w in cat_weapons if w['Layer'] == layer]
        if not layer_weapons: continue
            
        md_content += f'### {layer}\n\n'
        md_content += '| Weapon | Old Damage | AoS Damage | Sphere Speed | Old Speed | Swing (100 Dex) |\n'
        md_content += '| :--- | :--- | :--- | :--- | :--- | :--- |\n'
        
        for w in layer_weapons:
            old_dmg = f"{w['OldMin']}-{w['OldMax']}"
            aos_dmg = f"{w['AosMin']}-{w['AosMax']}"
            swing = f"{w['SwingSeconds']:.2f}s".replace('.', ',')
            md_content += f"| {w['Name']} | {old_dmg} | {aos_dmg} | {w['SphereSpeed']} | {w['OldSpeed']} | {swing} |\n"
            
        md_content += '\n'

with open('c:\\Users\\menti\\Desktop\\novo-shard\\WeaponCalc\\Weapon_Analysis_Report.md', 'w', encoding='utf-8') as f:
    f.write(md_content)

File: update.js
const fs = require('fs');
const data = JSON.parse(fs.readFileSync('Weapon_Analysis_Report.json', 'utf8').replace(/^\uFEFF/, ''));

data.sort((a, b) => a.Name.localeCompare(b.Name));

let md = '# Weapon Analysis Report\n\n';
const cats = ['Swords', 'Maces', 'Fencing', 'Bows', 'Throwing'];

cats.forEach(cat => {
    let catWeapons = data.filter(w => w.Skill === cat);
    if (!catWeapons.length) return;

    md += `## ${cat}\n\n`;

    ['OneHanded', 'TwoHanded'].forEach(layer => {
        let layerWeapons = catWeapons.filter(w => w.Layer === layer);
        if (!layerWeapons.length) return;

        md += `### ${layer}\n\n`;
        md += '| Weapon | Old Damage | AoS Damage | Sphere Speed | Old Speed | Swing (100 Dex) |\n';
        md += '| :--- | :--- | :--- | :--- | :--- | :--- |\n';

        layerWeapons.forEach(w => {
            let oldDmg = `${w.OldMin}-${w.OldMax}`;
            let aosDmg = `${w.AosMin}-${w.AosMax}`;
            let swing = `${w.SwingSeconds.toFixed(2)}s`.replace('.', ',');
            md += `| ${w.Name} | ${oldDmg} | ${aosDmg} | ${w.SphereSpeed} | ${w.OldSpeed} | ${swing} |\n`;
        });
        md += '\n';
    });
});

fs.writeFileSync('Weapon_Analysis_Report.md', md, 'utf8');
console.log('Done!');
